Editorial (#243)
openid/OpenID4VCI: openid-4-verifiable-credential-issuance-1_0.md

  * `format`: REQUIRED. A JSON string identifying the format of this Credential, i.e., `jwt_vc_json` or `ldp_vc`. Depending on the format value, the object contains further elements defining the type and (optionally) particular claims the Credential MAY contain and information about how to display the Credential. (#format-profiles) contains Credential Format Profiles introduced by this specification.
  * `scope`: OPTIONAL. A JSON string identifying the scope value that this Credential Issuer supports for this particular Credential. The value can be the same across multiple `credential_configurations_supported` objects. The Authorization Server MUST be able to uniquely identify the Credential Issuer based on the scope value. The Wallet can use this value in the Authorization Request as defined in (#credential-request-using-type-specific-scope). Scope values in this Credential Issuer metadata MAY duplicate those in the `scopes_supported` parameter of the Authorization Server.

>   * `cryptographic_binding_methods_supported`: OPTIONAL. Array of case sensitive strings that identify how the Credential is bound to the identifier of the End-User who possesses the Credential, as defined in (#credential-binding). Support for keys in JWK format [@!RFC7517] is indicated by the value `jwk`. Support for keys expressed as a COSE Key object [@!RFC8152] (for example, used in [@!ISO.18013-5]) is indicated by the value `cose_key`. When the Cryptographic Binding Method is a DID, valid values are a `did:` prefix followed by a method-name using a syntax as defined in Section 3.1 of [@!DID-Core], but without a `:`and method-specific-id. For example, support for the DID method with a method-name "example" would be represented by `did:example`. Support for all DID methods listed in Section 13 of [@DID_Specification_Registries] at that moment in time is indicated by sending a value `did` without any method-name.

Author: yes, this was added because there were implementers who asked for it. For example, those who integrate universal resolver code into their application can instantly handle all the DID methods supported by the universal resolver. If an implementer implements 1-5 DID methods, they can list all those, but if there are 20+, they don't want to.

Honestly, I don't think anyone wants/will to support all of the DID methods in the DID registry, but there is a need for a parameter for those who support "all of the most used DID methods" and we can ask people to define alias values that stand for a specific combination of certain DID methods, but that is not very realistic, so `did` is probably most realistic, usable thing.

cc @nklomp

>   * `cryptographic_binding_methods_supported`: OPTIONAL. Array of case sensitive strings that identify how the Credential is bound to the identifier of the End-User who possesses the Credential, as defined in (#credential-binding). Support for keys in JWK format [@!RFC7517] is indicated by the value `jwk`. Support for keys expressed as a COSE Key object [@!RFC8152] (for example, used in [@!ISO.18013-5]) is indicated by the value `cose_key`. When the Cryptographic Binding Method is a DID, valid values are a `did:` prefix followed by a method-name using a syntax as defined in Section 3.1 of [@!DID-Core], but without a `:`and method-specific-id. For example, support for the DID method with a method-name "example" would be represented by `did:example`. Support for all DID methods listed in Section 13 of [@DID_Specification_Registries] at that moment in time is indicated by sending a value `did` without any method-name.

Reviewer: IMO the generic `did` support will remain problematic, even now it is restricted to the spec registry. You will always have parties on any side that do not implement everything like @jogu mentioned, or which think they implement all of it, but the spec registry changed in the mean time.
It would be very wise to only allow specific did methods. We have seen in the wild, that implementers just choose  `did` because the spec allows for it and it is easy, leading to interop issues

>   * `cryptographic_binding_methods_supported`: OPTIONAL. Array of case sensitive strings that identify how the Credential is bound to the identifier of the End-User who possesses the Credential, as defined in (#credential-binding). Support for keys in JWK format [@!RFC7517] is indicated by the value `jwk`. Support for keys expressed as a COSE Key object [@!RFC8152] (for example, used in [@!ISO.18013-5]) is indicated by the value `cose_key`. When the Cryptographic Binding Method is a DID, valid values are a `did:` prefix followed by a method-name using a syntax as defined in Section 3.1 of [@!DID-Core], but without a `:`and method-specific-id. For example, support for the DID method with a method-name "example" would be represented by `did:example`. Support for all DID methods listed in Section 13 of [@DID_Specification_Registries] at that moment in time is indicated by sending a value `did` without any method-name.

Reviewer: Also people using the Universal Resolver, should be running their own instance anyway, and they really should restrict it to a few DID methods, as many DID specs/implementations are pretty poor

>   * `cryptographic_binding_methods_supported`: OPTIONAL. Array of case sensitive strings that identify how the Credential is bound to the identifier of the End-User who possesses the Credential, as defined in (#credential-binding). Support for keys in JWK format [@!RFC7517] is indicated by the value `jwk`. Support for keys expressed as a COSE Key object [@!RFC8152] (for example, used in [@!ISO.18013-5]) is indicated by the value `cose_key`. When the Cryptographic Binding Method is a DID, valid values are a `did:` prefix followed by a method-name using a syntax as defined in Section 3.1 of [@!DID-Core], but without a `:`and method-specific-id. For example, support for the DID method with a method-name "example" would be represented by `did:example`. Support for all DID methods listed in Section 13 of [@DID_Specification_Registries] at that moment in time is indicated by sending a value `did` without any method-name.

Author: ```suggestion
  * `cryptographic_binding_methods_supported`: OPTIONAL. Array of case sensitive strings that identify how the Credential is bound to the identifier of the End-User who possesses the Credential, as defined in (#credential-binding). Support for keys in JWK format [@!RFC7517] is indicated by the value `jwk`. Support for keys expressed as a COSE Key object [@!RFC8152] (for example, used in [@!ISO.18013-5]) is indicated by the value `cose_key`. When the Cryptographic Binding Method is a DID, valid values are a `did:` prefix followed by a method-name using a syntax as defined in Section 3.1 of [@!DID-Core], but without a `:`and method-specific-id. For example, support for the DID method with a method-name "example" would be represented by `did:example`.
```

          * `locale`: OPTIONAL. String value that identifies language of this object represented as language tag values defined in BCP47 [@!RFC5646]. There MUST be only one object for each language identifier.
* `order`: OPTIONAL. Array of the claim name values that lists them in the order they should be displayed by the Wallet.

> It is recommended to define an `@context` value to communicate additional information such as which claims are mandatory-to-be-issued, type of claim value (i.e., string, number, etc.), the display properties of a Credential, and the order of the claim values when displayed.

Reviewer 2: I don't understand the reason for this recommendation or I don't understand what this recommendation is trying to tell the reader to do; it says the purpose of defining a `@context` value is "to communicate additional information" but all the information listed can, as far as I can see, be communicated in the issuer metadata. Is it saying you should define which claims are mandatory both in an @context and in the issuer metadata?

I'm not really familiar with `@context`, does the example in this section follow that recommendation? It doesn't appear to as https://www.w3.org/2018/credentials/examples/v1 doesn't define any display properties I think.

Author: when @context is used, I would imagine that would take precedence over issuer metadata regarding type of claim value, mandatory or not, etc. not sure about display and order. but honestly, I am ok deleting it.

```suggestion
```

* `credential_definition`: OPTIONAL. Object containing the detailed description of the Credential consisting of the following parameter:
    * `@context`: OPTIONAL. Array as defined in (#server-metadata-ldp-vc). It MUST only be present if the `format` claim is present in the root of the authorization details object. It MUST not be present otherwise. 
    * `type`: OPTIONAL. Array as defined in (#server-metadata-ldp-vc).  This claim contains the type values the Wallet requests authorization for at the Credential Issuer. MUST only be present if the `@context` claim is present. 

>     * `credentialSubject`: OPTIONAL. Object as defined in (#authorization-jwt-vc-json) excluding the `display` parameter.

Author: ```suggestion
    * `credentialSubject`: OPTIONAL. Object as defined in (#server-metadata-sd-jwt-vc) excluding the `display` and `value_type` parameters. `mandatory` parameter here is used by the Wallet to indicate to the Issuer that it only accepts Credential(s) issued with those claim(s).
```

The following additional claims are defined for authorization details of type `openid_credential` and this Credential format.

* `doctype`: OPTIONAL. String as defined in (#server-metadata-mso-mdoc). This claim contains the type value the Wallet requests authorization for at the Credential Issuer. It MUST only be present if the `format` claim is present. It MUST not be present otherwise. 

> * `claims`: OPTIONAL. Object as defined in (#server-metadata-mso-mdoc) excluding the `display` parameter.

Author: ```suggestion
* `claims`: OPTIONAL. Object as defined in (#server-metadata-sd-jwt-vc) excluding the `display` and `value_type` parameters. `mandatory` parameter here is used by the Wallet to indicate to the Issuer that it only accepts Credential(s) issued with those claim(s).
```

The following additional claims are defined for authorization details of type `openid_credential` and this Credential format.

* `vct`: REQUIRED. String as defined in (#server-metadata-sd-jwt-vc). This claim contains the type values the Wallet requests authorization for at the Credential Issuer. It MUST only be present if the `format` claim is present. It MUST not be present otherwise.

> * `claims`: OPTIONAL. An object as defined in (#server-metadata-sd-jwt-vc) excluding the `display` parameter.

Reviewer 2: As we're only excluding `display`, did we conclude that the `mandatory` and `value_type` do have meaning if passed in the authorization details?

Author: ```suggestion
* `claims`: OPTIONAL. An object as defined in (#server-metadata-sd-jwt-vc) excluding the `display` and `value_type` parameters. `mandatory` parameter here is used by the Wallet to indicate to the Issuer that it only accepts Credential(s) issued with those claim(s).
```
